<!-- -*- xml -*-

     This tutorial relies on a recorded video sequence where the
     camera is fixed in front of a cube. The model corresponding to
     this cube is given into the models/laas-box directoy of this
     package.

     See http://www.ros.org/wiki/visp_tracker/Tutorials for more
     information.

     This is an alternative version launching twice the same tracker.
     It illustrates the use of ROS namespace to track several objects
     at the same time.

  -->

<launch>
  <!-- FIXME: do not use the timestamps from the bag file,
    it makes the tracker hang for some reason. -->
  <arg name="use_sim_time" default="false" />
  <executable cmd="ros2 bag play install/visp_tracker/share/bag/tutorial-static-box-ros2 --loop"/>

  <group>
    <node pkg="visp_tracker" name="visp_tracker_mbt" exec="visp_tracker_mbt">
      <param name="use_sim_time" value="$(var use_sim_time)" />
      <param name="camera_prefix" value="/wide_left/camera" />
      <param name="child_frame_id" value="box1" />
    </node>

    <node pkg="visp_tracker" name="visp_tracker_client" exec="visp_tracker_client">
      <param name="model_path" value="package://visp_tracker/models" />
      <param name="model_name" value="laas-box" />
      <param name="use_sim_time" value="$(var use_sim_time)" />
      <param from="package://visp_tracker/models/laas-box/tracker.yaml" />
    </node>

    <node pkg="visp_tracker" name="visp_tracker_viewer" exec="visp_tracker_viewer">
      <param name="use_sim_time" value="$(var use_sim_time)" />
      <param name="frame_size" value="0.2" />
      <param name="tracker_name" value="visp_tracker_client" />
    </node>

  </group>


  <group>
    <node pkg="visp_tracker" name="visp_tracker_mbt" exec="visp_tracker_mbt">
      <param name="camera_prefix" value="/wide_left/camera" />
      <param name="child_frame_id" value="box2" />
      <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>

    <node pkg="visp_tracker" name="visp_tracker_client" exec="visp_tracker_client">
      <param name="model_path" value="package://visp_tracker/models" />
      <param name="model_name" value="laas-box" />
      <param from="package://visp_tracker/models/laas-box/tracker.yaml" />
      <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>

    <node pkg="visp_tracker" name="visp_tracker_viewer" exec="visp_tracker_viewer">
      <param name="frame_size" value="0.2" />
      <param name="tracker_name" value="visp_tracker_client" />
      <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>
  </group>
</launch>
